### PR TITLE
Feature mtc 227 ensure messages traceable

### PR DESCRIPTION
**Pull request by GSailen**
Do not merge until tested

### CODE REVIEW

File: src/MarginTrading.SettingsService/Extensions/TraceabilityApiExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            if (traceability == null)
+            {
+                throw new ArgumentNullException(nameof(traceability));
+            }   
+        }
+        
+        public static string ExtractCorrelationId(this TraceableRequestParams @params)
+        {
+            return string.IsNullOrWhiteSpace(@params.CorrelationId)
+                ? @params.Id
```
Comment (cameronfletcher) [inline src/MarginTrading.SettingsService/Extensions/TraceabilityApiExtensions.cs]: I think this is wrong and may lead to confusion. Same for causation ID below. Happy to discuss.

File: src/MarginTrading.SettingsService.Contracts/Common/TraceableMessageBase.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        [Key(2)]
+        public string CausationId { get; set; }
+        
+        /// <summary>
+        /// Event creation time stamp in UTC time.
+        /// </summary>
+        [Key(3)]
+        public DateTime EventTimestamp { get; set; }
+
+        public TraceableMessageBase([NotNull] string correlationId, [CanBeNull] string causationId,
```
Comment (cameronfletcher) [inline src/MarginTrading.SettingsService.Contracts/Common/TraceableMessageBase.cs]: Correlation ID _may_ be null when the calling client (of, for instance, an API) does not supply one. In this instance one should be generated.

**Comment by cameronfletcher** [comment]
These new fields need to be logged in order for us to benefit from them.

**Comment by tarurar** [comment]
Outdated

**Pull request closed without merging**